src/message_pool/msgpool/msg_pool.rs

```diff
@@ -68,67 +72,72 @@ impl MsgSet {
         MsgSet {
             msgs: HashMap::new(),
             next_sequence: sequence,
+            required_funds: TokenAmount::from_atto(0),
         }
     }
 
     /// Add a signed message to the `MsgSet`. Increase `next_sequence` if the
     /// message has a sequence greater than any existing message sequence.
-    /// Use this method when pushing a message coming from trusted sources.
-    pub fn add_trusted<T>(&mut self, api: &T, m: SignedMessage) -> Result<(), Error>
-    where
-        T: Provider,
-    {
-        self.add(api, m, true)
-    }
-
-    /// Add a signed message to the `MsgSet`. Increase `next_sequence` if the
-    /// message has a sequence greater than any existing message sequence.
-    /// Use this method when pushing a message coming from untrusted sources.
-    #[allow(dead_code)]
-    pub fn add_untrusted<T>(&mut self, api: &T, m: SignedMessage) -> Result<(), Error>
-    where
-        T: Provider,
-    {
-        self.add(api, m, false)
-    }
-
-    fn add<T>(&mut self, api: &T, m: SignedMessage, trusted: bool) -> Result<(), Error>
+    pub fn add<T>(
+        &mut self,
+        api: &T,
+        m: SignedMessage,
+        strict: bool,
+        trusted: bool,
+    ) -> Result<(), Error>
     where
         T: Provider,
     {
-        let max_actor_pending_messages = if trusted {
-            api.max_actor_pending_messages()
+        let mut next_nonce = self.next_sequence;
```

Comment:
To match https://github.com/filecoin-project/lotus/blob/master/chain/messagepool/messagepool.go#L237